python/GafferSceneTest/InstancerTest.py

```diff
@@ -2360,6 +2364,62 @@ def testContextSetPerfWithVariationsSingleEvaluate( self ):
 	def testContextSetPerfWithVariationsParallelEvaluate( self ):
 		self.runTestContextSetPerf( True, True )
 
+	def initSimpleInstancer( self ):
+		mesh = IECoreScene.MeshPrimitive.createPlane(
+			imath.Box2f( imath.V2f( -1 ), imath.V2f( 1 ) ),
+			imath.V2i( 2499 )
+		)
+
+		mesh["index"] = IECoreScene.PrimitiveVariable(
+			IECoreScene.PrimitiveVariable.Interpolation.Vertex,
+			IECore.IntVectorData( [ 0, 1 ] * 3125000 ),
+		)
+
+		nodes = {}
+		nodes["meshSource"] = GafferScene.ObjectToScene()
+		nodes["meshSource"]["name"].setValue( "plane" )
+		nodes["meshSource"]["object"].setValue( mesh )
+
+		# Pre-evaluate the big object, so we just measure the instancer time
+		nodes["meshSource"]["out"].object( "/plane" )
+
+		nodes["sphere"] = GafferScene.Sphere()
+		nodes["cube"] = GafferScene.Cube()
+
+		nodes["parent"] = GafferScene.Parent()
+		nodes["parent"]["parent"].setValue( '/' )
+		nodes["parent"]["in"].setInput( nodes["sphere"]["out"] )
+		nodes["parent"]["children"][0].setInput( nodes["cube"]["out"] )
+
+
+		# Instancer
+		nodes["instancerFilter"] = GafferScene.PathFilter()
+		nodes["instancerFilter"]["paths"].setValue( IECore.StringVectorData( [ '/plane' ] ) )
+
+		nodes["instancer"] = GafferScene.Instancer()
+		nodes["instancer"]["in"].setInput( nodes["meshSource"]["out"] )
+		nodes["instancer"]["filter"].setInput( nodes["instancerFilter"]["out"] )
+		nodes["instancer"]["prototypes"].setInput( nodes["parent"]["out"] )
+		nodes["instancer"]["prototypeIndex"].setValue( "index" )
+
+		return nodes
+
+	@GafferTest.TestRunner.PerformanceTestMethod()
+	def testEngineDataPerf( self ):
+		Gaffer.ValuePlug.clearCache()
+		Gaffer.ValuePlug.clearHashCache( True )
+		nodes = self.initSimpleInstancer()
+		with GafferTest.TestRunner.PerformanceScope() :
+			self.assertEqual( nodes["instancer"]["out"].childNames( "/plane/instances" ), IECore.InternedStringVectorData( [ "sphere", "cube" ] ) )
+
+
+	@GafferTest.TestRunner.PerformanceTestMethod()
+	def testChildNamesPerf( self ):
```

Comment:
I see similar numbers:

```
Then
====
testEngineDataPerf : 0.000257s
testChildNamesPerf  : 0.992s

Now
===

testEngineDataPerf : 0.0791s
testChildNamesPerf : 0.924s
```

( Also with the 3rd run of testChildNamesPerf consistently being about 3X faster )

I'm not seeing as big an improvement in testChildNamesPerf. But yes, the numbers for testEngineDataPerf are expected - I thought it was worth documenting that more work has been moved into the EngineData constructor, but the only way this would be observed in practice is if you were only evaluating the list of prototypes ( ie. the children of "/instances" ), but not evaluating anything deeper in the hierarchy. testChildNamesPerf is a more realistic test.